modules/course_management/cog.py

```diff
@@ -77,23 +86,97 @@ async def add_course(
 		guild_ids=[config.guild_id],
 		options=[
 			create_option(
-				name="channel",
-				description="Mention a course the professor teaches. (eg. #automata)",
+				name="course",
+				description=(
+					"The channel corresponding to the course you want to delete."
+				),
 				option_type=SlashCommandOptionType.CHANNEL,
 				required=True,
 			),
 		],
+		connector={"course": "channel"},
 	)
 	@has_permissions(manage_channels=True)
 	async def delete_course(self, ctx: SlashContext, channel: discord.TextChannel):
-		await course_deleter.delete_course(ctx, channel.id)
+		await ctx.defer()
+		await course_deleter.delete_course(ctx, channel)
 		await ctx.send(
 			embed=embed_success(
 				"Well done... All evidence of that course has been deleted from the"
 				" face of the earth. Hope you're proud of yourself."
 			)
 		)
 
+	@cog_ext.cog_subcommand(
+		base="course",
+		name="activate",
+		description="Move an inactive course channel to the active courses list.",
+		guild_ids=[config.guild_id],
+		options=[
+			create_option(
+				name="course",
+				description=(
+					"The channel corresponding to the course you want to activate."
+				),
+				option_type=SlashCommandOptionType.CHANNEL,
+				required=True,
+			)
+		],
+	)
+	async def activate_course(self, ctx: SlashContext, course: discord.TextChannel):
+		await ctx.defer()
+		await course_activator.activate_course(ctx, course)
+		await ctx.send(embed=embed_success(f"Successfully activated #{course.name}."))
+
+	@cog_ext.cog_subcommand(
+		base="course",
+		name="deactivate",
+		description="Move an active course channel to the inactive courses list.",
+		guild_ids=[config.guild_id],
+		options=[
+			create_option(
+				name="course",
+				description=(
+					"The channel corresponding to the course you want to deactivate."
+				),
+				option_type=SlashCommandOptionType.CHANNEL,
+				required=True,
+			)
+		],
+	)
+	@has_permissions(manage_channels=True)
+	async def deactivate_course(self, ctx: SlashContext, course: discord.TextChannel):
+		await ctx.defer()
+		await course_activator.deactivate_course(ctx, course)
+		await ctx.send(embed=embed_success(f"Successfully deactivated #{course.name}."))
+
+	@cog_ext.cog_subcommand(
+		base="course",
+		name="deactivate-all",
+		description="Move all active course channels to the inactive courses list.",
+		guild_ids=[config.guild_id],
+	)
+	@has_permissions(manage_channels=True)
+	async def deactivate_all_courses(self, ctx: SlashContext):
+		await ctx.defer()
+		await course_activator.deactivate_all_courses(ctx)
+		await ctx.send(embed=embed_success(f"Successfully deactivated all courses."))
+
+	@cog_ext.cog_subcommand(
+		base="course",
+		name="sort",
+		description="Sort all course channels alphabetically.",
+		guild_ids=[config.guild_id],
+	)
+	@has_permissions(manage_channels=True)
+	async def sort_courses(self, ctx: SlashContext):
+		ctx.defer()
```

Comment:
I'm not sure that this is required if there is no response sent.

```suggestion
```

Reply:
Oh I think I forgot to add a response